docs/user_guide/mitigation.rst

```diff
@@ -80,11 +81,408 @@ Reductions
+>>> reduction.predict(X_test)  # doctest: +SKIP
+
+Fairlearn doesn't impose restrictions on the referenced :code:`estimator`
+other than the existence of :code:`fit` and :code:`predict` methods.
+At the moment, the :code:`estimator`'s :code:`fit` method also needs to
+provide a :code:`sample_weight` argument which the reductions techniques use
+to reweight samples.
+In the future Fairlearn will provide functionality to handle this even
+without a :code:`sample_weight` argument.
+
+Before looking more into the reductions approaches themselves this section
+examines the different constraints that are compatible with them. All of them
+are expressed as moments inheriting from :code:`ClassificationMoment` or
+:code:`LossMoment`, which in turn inherit from the base class :code:`Moment`.
+:code:`Moment`'s main purpose is to calculate the constraint violation of a
+current set of predictions through its :code:`gamma` function as well as to
+provide :code:`signed_weights` that are used to relabel and reweight samples.
+
+.. _constraints_binary_classification:
+
+Constraints for binary classification
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All existing constraints for binary classification inherit from
+:code:`ClassificationMoment` and can be formulated as difference or ratio-based
+constraints.
+
+Difference-based constraints consider the difference with respect
+to a metric between a group of samples and the overall population or
+cohorts thereof.
+Parity represents the state where
+
+.. math::
+
+    \text{metric}_a - \text{metric}_* = 0 \; \forall a
+
+where :math:`a` is a group indentifier and :math:`*` represents the overall
+population. This is internally represented as two inequalities
+
+.. math::
+
+    \text{metric}_a - \text{metric}_* \leq 0 \; \forall a \\
+    -\text{metric}_a + \text{metric}_* \leq 0 \; \forall a
+
+The algorithms do not consider every kind of violation of these constraints,
+but rather violations that go beyond their :code:`difference_bound`. Such
+a difference-based constraint could be instantiated as
+
+    >>> ClassificationMoment(difference_bound=0.01)  # doctest: +SKIP
+
+Note that achieving parity with the :code:`difference_bound` does not mean
+that the difference between the group with the highest metric value and the
+group with the lowest metric value is below the :code:`difference_bound`, but
+rather the difference between each individual group and the metric value for
+the overall population (or cohort thereof) is below :code:`difference_bound`. 
+
+In constrast, ratio-based constraints consider the ratio as opposed to the
+difference. Parity is represented as
+
+.. math::
+
+    r \leq \dfrac{\text{metric}_a}{\text{metric}_*} \leq \dfrac{1}{r} \; \forall a
+
+with a :code:`ratio_bound` :math:`r`. Similar to the :code:`difference_bound`
+for difference-based constraints, ratio-based constraints have a
+:code:`ratio_bound_slack`. The two inequalities can also be considered
```

Comment:
I think it would be helpful to show the slack for the ratios. I can think of a couple of possible ways it might appear, and I'm not sure which is correct.

Reply:
If you're not going to show it, I don't see why you'd mention it at this point. In the difference version, doesn't it change the zeros on the right hand side into epsilons?

Reply:
I'm showing it because I'm describing the moment objects. If I don't provide a constructor arg then I can't instantiate it (technically no arg = difference bound, so it would work but then I'm just hiding and confusing). I'm trying to help users understand what these are about, and how they relate to metrics (which are explained in the previous section). I can show mitigations in the mitigation technique section. Why should I show it in both? Doesn't quite make sense to me. 

We need to generally explain what the slack means at some point, but I don't want to block this PR on that. Documentation is more of an iterative process so we can come back to this.